src/userWallet.ts

```diff
@@ -95,7 +95,11 @@ export class UserWallet {
      * From an encrypted keyfile, given the password, loads the secret key and the public key.
      */
     static decryptSecretKey(keyFileObject: any, password: string): UserSecretKey {
-        // Here, we do not check the "kind" field. Older keystore files (holding only secret keys) do not have this field.
+        // Here, we check the "kind" field only for files that have it. Older keystore files (holding only secret keys) do not have this field.
+        const kind = keyFileObject.kind;
+        if (kind && kind !== UserWalletKind.SecretKey){
```

Comment:
I think it should have been `!=`. Interesting that it worked fine. Probably `UserWalletKind.SecretKey === "secretKey"` is evaluated to true. CC: @CiprianDraghici

Reply:
Indeed, both of the following output `true`:

```
console.log("A", UserWalletKind.SecretKey == "secretKey");
console.log("B", UserWalletKind.SecretKey === "secretKey");
```

:+1: